qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -609,6 +610,64 @@ def metadata_headers(cls):
             qdb.sql_connection.TRN.add(sql)
             return qdb.sql_connection.TRN.execute_fetchflatten()
 
+    def _common_delete_sample_steps(self, sample_name):
+        r"""Executes the common delete sample steps
+
+        Parameters
+        ----------
+        sample_name : str
+            The sample name to be erased
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the `sample_name` doesn't exist
+        """
+        if sample_name not in self.keys():
+            raise qdb.exceptions.QiitaDBColumnError(
+                "'%s' not in info file %d" % (sample_name, self._id))
+
+        with qdb.sql_connection.TRN:
+            sql = 'DELETE FROM qiita.{0}{1} WHERE sample_id=%s'.format(
+                self._table_prefix, self._id)
+            qdb.sql_connection.TRN.add(sql, [sample_name])
+
+            sql = "DELETE FROM qiita.{0} WHERE sample_id=%s".format(
+                    self._table)
+            qdb.sql_connection.TRN.add(sql, [sample_name])
+
+            qdb.sql_connection.TRN.execute()
+
+            self.generate_files()
+
+    def delete_column(self, column_name):
+        """Delete `column_name` from info file
+
+        Parameters
+        ----------
+        column : str
+            The column name to be deleted
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If a the info file can't be updated
+            If the `column_name` doesn't exist
+        """
+        if not self.can_be_updated(columns={column_name}):
+            raise qdb.exceptions.QiitaDBColumnError(
+                '%s cannot be deleted' % column_name)
+        if column_name not in self.categories():
```

Comment:
Hmmm, not very sure about this one, but I think the check to see if the column exists or not should come before the check to see if the column can be updated.